internal/middleware/correlation_id.go

```diff
@@ -0,0 +1,28 @@
+package middleware
+
+import (
+	"net/http"
+
+	"github.com/RHEnVision/provisioning-backend/internal/ctxval"
+)
+
+func CorrelationID(next http.Handler) http.Handler {
+	fn := func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+
+		// Edge request id
+		corrId := r.Header.Get("X-Correlation-Id")
+		if corrId != "" {
+			ctx = ctxval.WithCorrelationId(ctx, corrId)
+		}
+
+		// Store in response headers for easier debugging
+		w.Header().Set("X-Correlation-Id", corrId)
+		logger := ctxval.Logger(ctx).With().Str("correlation_id", corrId).Logger()
+		logger.Trace().Msgf("Added correlation id %s to logger", corrId)
```

Comment:
Can you move this into the if statement above? This should not be executed when no header is present, otherwise this will add the following (empty) header to every single request:

    X-Correlation-Id: